### PR TITLE
Issue 967

### PR DESCRIPTION
**Pull request by Naupio**
## What problem does this PR solve?
link issue: #967 
1. add token instance for save token metadata(token url/url)
2. update token instance when sync worker import token_transfer
3. add mix task to process history data
4. add global constants
5. add token instance field for erc721_inventory/erc1155_user_inventory/erc1155_inventory object

## Check List
1. check auto update token instance when sync worker import token_transfer
2. check history metadata is updated when mix task finished
3. check erc721_inventory/erc1155_user_inventory/erc1155_inventory  with token instance

#### Test
- unit test
#### Task
 - need process transaction history data

### CODE REVIEW

File: lib/godwoken_indexer/block/sync_worker.ex [hunk 1]
```diff
@@ -821,6 +825,36 @@ defmodule GodwokenIndexer.Block.SyncWorker do
...
+      |> Enum.map(fn tt ->
+        %{
+          "token_contract_address_hash" => tt.token_contract_address_hash,
+          "token_id" => tt.token_id
+        }
+      end)
+
+    if length(with_token_ids) > 0 do
+      Enum.each(with_token_ids, fn args -> ERC721ERC1155InstanceMetadata.new_job(args) end)
+    else
```
Comment (zmcNotafraid) [inline lib/godwoken_indexer/block/sync_worker.ex:853]: else 不写也没问题把？

File: lib/godwoken_indexer/worker/erc721_erc1155_instance_metadata.ex [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  def check_token_instance_retrieved(%{
+        "token_contract_address_hash" => token_contract_address_hash,
+        "token_id" => token_id
+      })
+      when is_bitstring(token_contract_address_hash) and is_integer(token_id) do
+    {:ok, token_contract_address_hash} = Address.cast(token_contract_address_hash)
+    token_id = Decimal.new(token_id)
+
+    with return when not is_nil(return) <-
+           Repo.get_by(TokenInstance,
```
Comment (zmcNotafraid) [inline lib/godwoken_indexer/worker/erc721_erc1155_instance_metadata.ex:116]: 这里这样一个个查效率太低，建议直接` where (token_contract_address_hash, token_id) IN ( (A, X), (B, Y), (C, Z) );` 统一提前过滤下

**Review by zmcNotafraid — CHANGES_REQUESTED** [review]

File: lib/godwoken_indexer/block/sync_worker.ex (hunk 1 above)
Comment (Naupio) [reply lib/godwoken_indexer/block/sync_worker.ex]: fixed

File: lib/godwoken_indexer/worker/erc721_erc1155_instance_metadata.ex (hunk 2 above)
Comment (Naupio) [reply lib/godwoken_indexer/worker/erc721_erc1155_instance_metadata.ex:163]: added precheck

File: lib/godwoken_indexer/worker/erc721_erc1155_instance_metadata.ex [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+  def token_instance_upsert(params) do
+    changeset = TokenInstance.changeset(%TokenInstance{}, params)
+
+    Repo.insert(changeset,
+      on_conflict: :replace_all,
+      conflict_target: [:token_id, :token_contract_address_hash]
+    )
+  end
+
+  def check_token_instance_retrieved(%{
```
Comment (zmcNotafraid) [inline lib/godwoken_indexer/worker/erc721_erc1155_instance_metadata.ex:144]: 这个方法没看到在哪使用？

**Review by zmcNotafraid — CHANGES_REQUESTED** [review]

File: lib/godwoken_indexer/worker/erc721_erc1155_instance_metadata.ex (hunk 3 above)
Comment (Naupio) [reply lib/godwoken_indexer/worker/erc721_erc1155_instance_metadata.ex]: deleted unused function

**Review by zmcNotafraid — APPROVED** [review]

**Review by FrederLu — APPROVED** [review]

**Pull request merged**